xarray/core/utils.py

```diff
@@ -787,6 +787,14 @@ def drop_dims_from_indexers(
         )
 
 
+class property_:
+    def __init__(self, func):
```

Comment:
let's add a docstring describing why this is necessary and what its used for.

Reply:
sure. Do you have a suggestion for a more descriptive name?

Reply:
Not really...

But lets mention in the docstring that we do this specifically to make Sphinx work.

Reply:
that's not the only thing it enables: without it, tab-completion on the class does not work either